### ISSUE TITLE
Generic.Formatting.MultipleStatementAlignment skipping assignments within closures

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1925]
Following up on #1870, #1848 and open PR #1924.

The fixes in the commits related to the above issues, all skip past closures and arrays.

While working on #1924, I realized however, that as the sniff returns a `stackPtr` to where the assignment block ended, this means that assignments within closures/anonymous classes will never be examined nor trigger an error.

For arrays, I'm not too concerned as the double arrow alignment is often covered by an array focused sniff anyway.

Code sample demonstrating the issue:
```php
$loggerResult = $util->setLogger(new class {
    public function log($msg)
    {
        $a = $msg;
        $foobar = $msg;
        $bar = $msg;
    }
});
$foo          = 5;
```

The assignments within the closure are not aligned, but will not trigger an error either.

**Comment by gsherwood** [comment]
Thanks for picking this up - it was a pain to fix, but I think I got it.

I had to revert the simpler fix for #1922 so that all the tokens would be checked, which allowed the sniff to recurse and check inside anon classes and closures. It's not very efficient, but the tests are all passing.

**Comment by jrfnl** [comment]
@gsherwood You're a star!

**Comment by jrfnl** [comment]
@gsherwood I think I'm seeing undesired side-effects of the fix.

Given this code:
```php
$abc = 'something';
if ( $something ) {
    // do something
} else {
   // do something else
}
$defghi = 'longer something;'
```

The sniff now appears to expects the assignment operators for the `$abc` and the `$defghi` variables to be aligned.
